### ISSUE TITLE
rotate_angle incorrect

### ISSUE THREAD

**Issue by gridley** [openmc-dev/openmc#1359]
Greetings peeps,

Seems that rotate_angle isn't working correctly. IIRC, the polar coordinates phi=0, theta=pi/2 should be a unit vector that point along the x axis, but in OpenMC, rotate_angle returns a unit vector that points along the -y axis for these polar inputs, using the reference (polar) axis as +z.

Should be an easy fix!

Check out this tiny example program if you don't believe me.

```cpp
#include "math_functions.h"
#include <iostream>

using namespace std;
using namespace openmc;

// Is this the expected direction result?

int main()
{
  Direction z_axis = {0., 0., 1.};
  double phi = 0.0;
  Direction should_be_x = rotate_angle(z_axis, 0., &phi);
  cout << "Components of theta=pi/2, phi=0:" << endl;
  cout << "x = " << should_be_x.x << endl;
  cout << "y = " << should_be_x.y << endl;
  cout << "z = " << should_be_x.z << endl;
}
```
Gives:

```
gavin@gpad:~/code/openmc$ g++ test.cpp -Iinclude/openmc build/lib/libopenmc.so 
gavin@gpad:~/code/openmc$ ./a.out 
Components of theta=pi/2, phi=0:
x = 0
y = -1
z = 0
```

Or, maybe our semantics about polar coordinates are different.

**Comment by smharper** [comment]
What application brought up this issue? Is it setting the angular distribution of a source for a fixed source calculation?

The only other use for `rotate_angle` that I know of is our collision physics. It works fine there because the azimuth reference direction is arbitrary in those cases. I get the argument for changing `rotate_angle` to make the fixed source setup match the conventions better, but I'm hesitant because it will break reproducibility of particle tracks from the current version. Maybe we could modify the way that source site sampling uses `rotate_angle`.

**Comment by makeclean** [comment]
That might also explain an oddity we found for the use of an angular dependent neutron source, which didn't behave as we expected. We presumed it was an error in our method to produce the source and figured we'd come back to it. I think certainly for sources this convention break is problematic.

When you say reproducibility of tracks, you mean pre and post 'fix' right?  I'm not fully caffeined yet, but I don't why that would be a problem, isit the tests?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by gridley** [comment]
@smharper Trying to set an angle-dependent source that is uniform over half the unit sphere, for setting up boundary conditions in collision probability calcs. I ended up subtracting pi/2 from all my phi coordinates and got what I needed. Why not just update the unit tests to match particle track reproducibility? Is that what you're referring to?

@makeclean glad to hear this may help you out!

**Comment by smharper** [comment]
@makeclean, right I'm talking about reproducing results pre- and post-fix.

It's no big deal for us to update the tests, but it's still annoying to our users when OpenMC gives different answers after an update (even if there's no statistically significant difference in those answers).

**Comment by paulromano** [comment]
This was fixed by #1363.
